sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusSenderClientTest.java

```diff
@@ -0,0 +1,160 @@
+    }
+
+    @BeforeEach
+    void setup() {
+        MockitoAnnotations.initMocks(this);
+        when(asyncSender.getEntityPath()).thenReturn(ENTITY_NAME);
+        when(asyncSender.getFullyQualifiedNamespace()).thenReturn(NAMESPACE);
+        sender = new ServiceBusSenderClient(asyncSender, RETRY_TIMEOUT);
+    }
+
+    @AfterEach
+    void teardown() {
+        sender.close();
+        singleMessageCaptor = null;
+        Mockito.framework().clearInlineMocks();
+    }
+
+    @Test
+    void verifyProperties() {
+        Assertions.assertEquals(ENTITY_NAME, sender.getEntityPath());
+        Assertions.assertEquals(NAMESPACE, sender.getFullyQualifiedNamespace());
+    }
+
+    /**
+     * Verifies that an exception is thrown when we create a batch with null options.
+     */
+    @Test
+    void createBatchNull() {
+        Assertions.assertThrows(NullPointerException.class, () -> sender.createBatch(null));
+    }
+
+    /**
+     * Verifies that the default batch is the same size as the message link.
+     */
+    @Test
+    void createBatchDefault() {
+        // Arrange
+        ServiceBusMessageBatch batch =  new ServiceBusMessageBatch(MAX_MESSAGE_LENGTH_BYTES, null, null,
+            null);
+        when(asyncSender.createBatch()).thenReturn(Mono.just(batch));
+
+        //Act
+        ServiceBusMessageBatch batchMessage = sender.createBatch();
+
+        //Assert
+        Assertions.assertEquals(MAX_MESSAGE_LENGTH_BYTES, batchMessage.getMaxSizeInBytes());
+        Assertions.assertEquals(0, batchMessage.getCount());
+        verify(asyncSender).createBatch();
+    }
+
+    /**
+     * Verifies we cannot create a batch if the options size is larger than the link.
+     */
+    @Test
+    void createBatchWhenSizeTooBigThanOnSendLink() {
+        // Arrange
+        int maxLinkSize = 1024;
+        int batchSize = maxLinkSize + 10;
+
+        // This event is 1024 bytes when serialized.
+        final CreateBatchOptions options = new CreateBatchOptions().setMaximumSizeInBytes(batchSize);
+        when(asyncSender.createBatch(options)).thenThrow(new IllegalArgumentException("too large size"));
+
+        // Act & Assert
+        Assertions.assertThrows(IllegalArgumentException.class, () -> sender.createBatch(options));
+        verify(asyncSender, times(1)).createBatch(options);
+    }
+
+    /**
+     * Verifies that the producer can create a batch with a given {@link CreateBatchOptions#getMaximumSizeInBytes()}.
+     */
+    @Test
+    void createsMessageBatchWithSize() {
+        // Arrange
+        int batchSize = 1024;
+
+        final CreateBatchOptions options = new CreateBatchOptions().setMaximumSizeInBytes(batchSize);
+        ServiceBusMessageBatch batch =  new ServiceBusMessageBatch(batchSize, null, null,
+            null);
+        when(asyncSender.createBatch(options)).thenReturn(Mono.just(batch));
+
+        // Act & Assert
+        ServiceBusMessageBatch messageBatch = sender.createBatch(options);
```

Comment:
In Arrange, Act, and Assert, this line would be "Act". And the Line 135 would be "Assert"